### DISCUSSION TITLE
removeparam starting with tilde without regex

### DISCUSSION THREAD

**Discussion by injust** [uBlockOrigin/uBlock-issues#3326] [Q&A] [answered]
I wrote something like `$removeparam=~foo`, where `~foo` is the literal parameter I want to remove. But that actually removes all parameters except for `foo`.

I know that `$removeparam=/^~foo=/` works, but can this be done without the regex form?

**Comment by uBlock-user** [comment]
`$removeparam=foo`

params always start with `?` or `&`, else it's invalid

**Reply by injust** [reply]
But that removes `foo`, not `~foo`.

**Reply by uBlock-user** [reply]
`~foo` translates to everything except `foo` it's not even a valid param

**Reply by injust** [reply]
Yes, that's basically my question -- is there any way to remove a literal `~foo` without using the regex form?

Like is there any way to escape the `~` or something? I tried using a backslash, but that didn't work.

**Reply by uBlock-user** [reply]
@injust post link where I can test for myself

**Reply by gorhill** [reply, marked as answer]
> is there any way to remove a literal `~foo` without using the regex form

You have to use the regex form for that case: `$removeparam=/^\x7efoo=/`

**Reply by injust** [reply]
@gorhill Great, thanks for confirming.

Is there a specific reason that you used `\x7e` instead of `~`? I didn't think `~` had any special meaning inside the regex.

**Reply by gorhill** [reply]
No special reason beside failing to think properly, you're right, `~` would have worked as well.

**Reply by garry-ut99** [reply]
This syntax works too: `$removeparam=|~foo=`

**Reply by gorhill** [reply]
`|` is deprecated though, support for this syntax will be removed eventually -- it's not part of the official documentation.

**Reply by garry-ut99** [reply]
Well, then as per descriptions of: https://github.com/gorhill/uBlock/commit/d1895d4749b76de5c06ed6545ac2a4a3cd1724a4 and https://github.com/gorhill/uBlock/commit/e08f8cb00126175a44d939bfdf48c98d27e2ccf2, it should have been removed a long time ago already, because the only commit which ever existed and added 2 `=|` filters: https://github.com/uBlockOrigin/uAssets/commit/3df9d3d930e1608e440812f0f05e3cc0166a22f6 (`11.2020`) was removed after 2 months in https://github.com/uBlockOrigin/uAssets/commit/c2e4fa1a9213a0b7d73ffb07c18071e17376d809 (`01.2021`). Since then, which means since 3.5 year,  no new `=|` filters appeared ever.
